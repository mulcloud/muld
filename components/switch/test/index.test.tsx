import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup, fireEvent } from '@testing-library/react';
import 'regenerator-runtime';
import Switch from '..';

describe('Switch', () => {
    afterEach(cleanup);
    it('disabled prop', () => {
        const { asFragment } = render(<Switch disabled />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('loading prop', () => {
        const { asFragment } = render(<Switch loading value={true} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('color prop', () => {
        const { asFragment } = render(
            <Switch activeColor="#07c160" inactiveColor="#ee0a24" value={true} />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it('input event & change event', () => {
        const onInput = jest.fn();
        const onChange = jest.fn();
        render(<Switch onChange={onChange} onInput={onInput} />);
        fireEvent.click(document.body.querySelector('.mul-switch__node')!);
        expect(onInput).toHaveBeenCalled();
        expect(onChange).toHaveBeenCalled();
    });
    it('input event & disabled', () => {
        const onInput = jest.fn();
        const onChange = jest.fn();
        render(<Switch onChange={onChange} onInput={onInput} disabled />);
        fireEvent.click(document.body.querySelector('.mul-switch__node')!);
        expect(onInput).toHaveBeenCalledTimes(0);
        expect(onChange).toHaveBeenCalledTimes(0);
    });
    it('activeValue & inactiveValue prop', () => {
        const onInput = jest.fn();
        const onChange = jest.fn();
        render(
            <Switch
                value="1"
                onChange={onChange}
                onInput={onInput}
                activeValue="1"
                inactiveValue="2"
            />,
        );
        fireEvent.click(document.body.querySelector('.mul-switch__node')!);
        expect(onInput).toHaveBeenCalledWith('2');
        expect(onChange).toHaveBeenCalledWith('2');
    });
});
