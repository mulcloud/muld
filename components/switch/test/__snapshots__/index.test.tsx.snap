// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch color prop 1`] = `
<DocumentFragment>
  .c0.mul-switch {
  position: relative;
  display: inline-block;
  box-sizing: content-box;
  width: 2em;
  height: 1em;
  font-size: 30px;
  background-color: #fff;
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 1em;
  cursor: pointer;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
}

.c0.mul-switch .mul-switch__node {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  width: 1em;
  height: 1em;
  background-color: #fff;
  border-radius: 100%;
  box-shadow: 0 3px 1px 0 rgba(0,0,0,0.05), 0 2px 2px 0 rgba(0,0,0,0.1),0 3px 3px 0 rgba(0,0,0,0.05);
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
  -webkit-transition: transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
  transition: transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
}

.c0.mul-switch .mul-switch__loading {
  top: 25%;
  left: 25%;
  width: 50%;
  height: 50%;
  line-height: 1;
}

.c0.mul-switch--on {
  background-color: #108ee9;
}

.c0.mul-switch--on .mul-switch__node {
  -webkit-transform: translateX(1em);
  -ms-transform: translateX(1em);
  transform: translateX(1em);
}

.c0.mul-switch--on .mul-switch__loading {
  color: #108ee9;
}

.c0.mul-switch--disabled {
  cursor: not-allowed!important;
  opacity: 0.5;
}

.c0.mul-switch--loading {
  cursor: default;
}

<div
    aria-checked="true"
    class="c0 mul-switch mul-switch--on"
    style="background-color: rgb(7, 193, 96);"
  >
    <div
      class="mul-switch__node"
    />
  </div>
</DocumentFragment>
`;

exports[`Switch disabled prop 1`] = `
<DocumentFragment>
  .c0.mul-switch {
  position: relative;
  display: inline-block;
  box-sizing: content-box;
  width: 2em;
  height: 1em;
  font-size: 30px;
  background-color: #fff;
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 1em;
  cursor: pointer;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
}

.c0.mul-switch .mul-switch__node {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  width: 1em;
  height: 1em;
  background-color: #fff;
  border-radius: 100%;
  box-shadow: 0 3px 1px 0 rgba(0,0,0,0.05), 0 2px 2px 0 rgba(0,0,0,0.1),0 3px 3px 0 rgba(0,0,0,0.05);
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
  -webkit-transition: transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
  transition: transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
}

.c0.mul-switch .mul-switch__loading {
  top: 25%;
  left: 25%;
  width: 50%;
  height: 50%;
  line-height: 1;
}

.c0.mul-switch--on {
  background-color: #108ee9;
}

.c0.mul-switch--on .mul-switch__node {
  -webkit-transform: translateX(1em);
  -ms-transform: translateX(1em);
  transform: translateX(1em);
}

.c0.mul-switch--on .mul-switch__loading {
  color: #108ee9;
}

.c0.mul-switch--disabled {
  cursor: not-allowed!important;
  opacity: 0.5;
}

.c0.mul-switch--loading {
  cursor: default;
}

<div
    aria-checked="false"
    class="c0 mul-switch mul-switch--disabled"
    style="background-color: rgb(255, 255, 255);"
  >
    <div
      class="mul-switch__node"
    />
  </div>
</DocumentFragment>
`;

exports[`Switch loading prop 1`] = `
<DocumentFragment>
  .c0.mul-switch {
  position: relative;
  display: inline-block;
  box-sizing: content-box;
  width: 2em;
  height: 1em;
  font-size: 30px;
  background-color: #fff;
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 1em;
  cursor: pointer;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
}

.c0.mul-switch .mul-switch__node {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  width: 1em;
  height: 1em;
  background-color: #fff;
  border-radius: 100%;
  box-shadow: 0 3px 1px 0 rgba(0,0,0,0.05), 0 2px 2px 0 rgba(0,0,0,0.1),0 3px 3px 0 rgba(0,0,0,0.05);
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
  -webkit-transition: transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
  transition: transform 0.3s cubic-bezier(0.3,1.05,0.4,1.05);
}

.c0.mul-switch .mul-switch__loading {
  top: 25%;
  left: 25%;
  width: 50%;
  height: 50%;
  line-height: 1;
}

.c0.mul-switch--on {
  background-color: #108ee9;
}

.c0.mul-switch--on .mul-switch__node {
  -webkit-transform: translateX(1em);
  -ms-transform: translateX(1em);
  transform: translateX(1em);
}

.c0.mul-switch--on .mul-switch__loading {
  color: #108ee9;
}

.c0.mul-switch--disabled {
  cursor: not-allowed!important;
  opacity: 0.5;
}

.c0.mul-switch--loading {
  cursor: default;
}

.c1.mul-loading {
  position: relative;
  color: #c8c9cc;
  font-size: 0;
  vertical-align: middle;
}

.c1.mul-loading .mul-loading__spinner {
  position: relative;
  display: inline-block;
  width: 30px;
  max-width: 100%;
  height: 30px;
  max-height: 100%;
  vertical-align: middle;
  -webkit-animation: mul-rotate 0.8s linear infinite;
  animation: mul-rotate 0.8s linear infinite;
}

.c1.mul-loading .mul-loading__spinner--spinner {
  -webkit-animation-timing-function: steps(12);
  animation-timing-function: steps(12);
}

.c1.mul-loading .mul-loading__spinner--spinner i {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c1.mul-loading .mul-loading__spinner--spinner i::before {
  display: block;
  width: 2px;
  height: 25%;
  margin: 0 auto;
  background-color: currentColor;
  border-radius: 40%;
  content: ' ';
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(1) {
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg);
  opacity: 1;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(2) {
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
  opacity: 0.9375;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(3) {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  opacity: 0.875;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(4) {
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  transform: rotate(120deg);
  opacity: 0.8125;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(5) {
  -webkit-transform: rotate(150deg);
  -ms-transform: rotate(150deg);
  transform: rotate(150deg);
  opacity: 0.75;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(6) {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  opacity: 0.6875;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(7) {
  -webkit-transform: rotate(210deg);
  -ms-transform: rotate(210deg);
  transform: rotate(210deg);
  opacity: 0.625;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(8) {
  -webkit-transform: rotate(240deg);
  -ms-transform: rotate(240deg);
  transform: rotate(240deg);
  opacity: 0.5625;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(9) {
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
  opacity: 0.5;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(10) {
  -webkit-transform: rotate(300deg);
  -ms-transform: rotate(300deg);
  transform: rotate(300deg);
  opacity: 0.4375;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(11) {
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
  opacity: 0.375;
}

.c1.mul-loading .mul-loading__spinner--spinner i:nth-of-type(12) {
  -webkit-transform: rotate(360deg);
  -ms-transform: rotate(360deg);
  transform: rotate(360deg);
  opacity: 0.3125;
}

.c1.mul-loading .mul-loading__spinner--circular {
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
}

.c1.mul-loading .mul-loading__circular {
  display: block;
  width: 100%;
  height: 100%;
}

.c1.mul-loading .mul-loading__circular circle {
  -webkit-animation: mul-circular 1.5s ease-in-out infinite;
  animation: mul-circular 1.5s ease-in-out infinite;
  stroke: currentColor;
  stroke-width: 3;
  stroke-linecap: round;
}

.c1.mul-loading .mul-loading__text {
  display: inline-block;
  margin-left: 8px;
  color: #969799;
  font-size: 14px;
  vertical-align: middle;
}

.c1.mul-loading .mul-loading--vertical {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.mul-loading .mul-loading--vertical .mul-loading__text {
  margin: 8px 0 0;
}

<div
    aria-checked="true"
    class="c0 mul-switch mul-switch--on mul-switch--loading"
    style="background-color: rgb(25, 137, 250);"
  >
    <div
      class="mul-switch__node"
    >
      <div
        class="c1 mul-loading mul-loading--circular"
      >
        <span
          class="mul-loading__spinner mul-loading__spinner--circular"
          style="color: rgb(25, 137, 250);"
        >
          <svg
            class="mul-loading__circular"
            viewBox="25 25 50 50"
          >
            <circle
              cx="50"
              cy="50"
              fill="none"
              r="20"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
